near-sdk/src/json_types/account.rs

```diff
@@ -33,35 +43,47 @@ impl AsRef<AccountId> for ValidAccountId {
     }
 }
 
-impl<'de> serde::Deserialize<'de> for ValidAccountId {
-    fn deserialize<D>(deserializer: D) -> Result<Self, <D as serde::Deserializer<'de>>::Error>
+impl<'de> Deserialize<'de> for ValidAccountId {
+    fn deserialize<D>(deserializer: D) -> Result<Self, <D as de::Deserializer<'de>>::Error>
     where
-        D: serde::Deserializer<'de>,
+        D: de::Deserializer<'de>,
     {
-        let s = <String as serde::Deserialize>::deserialize(deserializer)?;
-        s.try_into()
-            .map_err(|err: Box<dyn std::error::Error>| serde::de::Error::custom(err.to_string()))
+        let s: String = Deserialize::deserialize(deserializer)?;
+        Self::try_from(s).map_err(de::Error::custom)
```

Comment:
```suggestion
        s.parse().map_err(de::Error::custom)
```

I imagine we want to remove TryFrom one day

Reply:
I kept it here because we have a string, and we don't want to call `FromStr` because it forces the string to be reallocated after. 

Sure, this clone might be optimized out but I was airing on the side of not having that be possible for now.

Reply:
Ah, yeah, you are absolutely correct!